WordPress/gutenberg
# [Mobile] - RichText - Use parsed font size values when comparing new changes

- Gutenberg Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4463

## Description
This PR fixes a bug within `RichText`, where in some cases it would generate a render loop when setting a new font-size value and this didn't match the parsed value.

With this change it'll parse the `currentFontSizeStyle` and `prevFontSizeStyle` before comparing values to update the state. As well as **not using** the font size value from its props if there's a font size provided from the styles.

**Note:** The font size picker is behind the `__DEV__` flag so it'll only show up with metro running for block-based themes. But **it will render** the font-size value correctly for both default and block-based themes.

## How has this been tested?

### Test case 1
- Open the app
- Switch to the HTML mode
- Set this content

```

<p class="has-text-color" style="color:#fcb900;font-size:35.56px">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed imperdiet ut nibh vitae ornare. Sed auctor nec augue at blandit.</p>

```
- Go back to the visual mode
- Expect the `Paragraph` block to show correctly.

### Test case 2

**Precondition**: Using a block-based theme.

- Open the WordPress app
- Start a new post
- Within the `Paragraph` block type some text
- Open the block settings
- Change the font size using the font size picker
- **Expect** to see the new selected font size.

### Test case 3

**Precondition**: Using a block-based theme.

- Open the WordPress app
- Start a new post
- Add a `Heading` block and type some text
- Change between different heading levels and **expect** the size to change correctly
- Change the font size using the font picker and **expect** the size to change correctly

## Screenshots 

Test case 1 | Test case 2 | Test case 3
-|-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/149367883-8074ac91-eaa4-42b0-be55-e629457f6f53.png" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/149518240-0011de1a-d910-4854-bbe1-d9932666da5a.gif" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/149518247-e3c88ad7-13be-4904-a937-8c5d84f80a8c.gif" width="200" /></kbd>

## Types of changes
Bug fix

## Checklist:
